src/components/select-field/BaseSelectField.js

```diff
@@ -321,12 +330,8 @@ class BaseSelectField extends React.Component<Props, State> {
 
                     this.selectOption(index);
                 },
-                /* preventDefault on mousedown so blur doesn't happen before click */
-                onMouseDown: event => {
```

Comment:
This was causing the issue where clicking on the scrollbar (when overlay has max height) would close the dropdown. Setting the mouse down event on the overlay itself achieves the same result without causing the issue